### PR TITLE
Adding abort method in my.request Promise. Query now accept optional params.

### PR DESCRIPTION
**Pull request by AlbertSabate**

### CODE REVIEW

File: types/api/network.d.ts [hunk 1]
```diff
@@ -88,7 +95,7 @@ declare namespace my {
   /**
    * 小程序网络请求
    */
-  function request(options: IHttpRequestOptions): void;
+  function request(options: IHttpRequestOptions): IRequestPromise;
```
Comment (yibuyisheng) [inline types/api/network.d.ts]: In the [document](https://opendocs.alipay.com/mini/api/owycmh#mPip1), the return type of `request` called `RequestTask`.

File: types/app.d.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 declare namespace tinyapp {
-  type Query = Record<string, string | number>;
+  type Query = Record<string, string | number | undefined>;
```
Comment (yibuyisheng) [inline types/app.d.ts]: `Query` is deserialized from 'URL', so I think `undefined` is not the possible value.

**Review by yibuyisheng — CHANGES_REQUESTED** [review]

File: types/api/network.d.ts (hunk 1 above)
Comment (AlbertSabate) [reply types/api/network.d.ts]: Changes on this part updated.

File: types/app.d.ts (hunk 2 above)
Comment (AlbertSabate) [reply types/app.d.ts]: We have the case where we use context to access to a page, which depends on the context and then we need 1 extra query param.

example:
```
interface PageChatQuery extends tinyapp.Query {
  context: 'agent' | 'driver';
  bookingId?: string;
}
```

In this case, we can deserialize the query, but we need to control the undefined in our code. Do you have any suggestion for this case without use undefined?

Thanks!!!! :)

Comment (AlbertSabate) [reply types/app.d.ts]: Ok, we solve it with conditional types, not bad though. Check the code, it is updated!

**Comment by yibuyisheng** [comment]
Thanks for your contribution.😄

**Pull request merged**